core/lib/contract_verifier/src/etherscan/client.rs

```diff
@@ -0,0 +1,329 @@
+use std::{collections::HashMap, time::Duration};
+
+use reqwest::Client;
+use serde::Serialize;
+use zksync_types::Address;
+
+use super::{
+    errors::{is_blocked_by_cloudflare_response, is_cloudflare_security_challenge, ApiError},
+    types::{ApiAction, ApiModule, ApiRequest, ApiResponseRaw, EtherscanVerificationRequest},
+};
+
+#[derive(Debug, Clone)]
+pub(crate) struct EtherscanClient {
+    api_url: String,
+    api_key: String,
+    http_client: Client,
+}
+
+fn process_api_call_result(
+    result: Result<impl AsRef<str>, reqwest::Error>,
+) -> Result<String, ApiError> {
+    match result {
+        Ok(result) => {
+            let result = result.as_ref();
+            let raw_response: ApiResponseRaw = serde_json::from_str(result).map_err(|error| {
+                if result.contains("Page not found") {
+                    ApiError::PageNotFound
+                } else if is_blocked_by_cloudflare_response(result) {
```

Comment:
I guess it also should have [non-200 error code](https://developers.cloudflare.com/support/troubleshooting/cloudflare-errors/troubleshooting-cloudflare-1xxx-errors/#errors-1006-1007-1008-or-1106-access-denied-your-ip-address-has-been-banned).